homeassistant/components/climate/systemair_savecair.py

```diff
@@ -0,0 +1,286 @@
+        if "supply_air_temp" not in self._client.data:
+            return None
+
+        air_temperature = self._client.data["supply_air_temp"]
+
+        if air_temperature is None:
+            return None
+
+        return air_temperature / 10
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        if "control_regulation_temp_unit" not in self._client.data:
+            return TEMP_CELSIUS
+
+        if self._client.data["control_regulation_temp_unit"] == "celsius":
+            return TEMP_CELSIUS
+
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if "pdm_input_temp_value" not in self._client.data:
+            return None
+
+        current_temperature = self._client.data["pdm_input_temp_value"]
+
+        if current_temperature is None:
+            return None
+
+        return current_temperature / 10
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if "main_temperature_offset" not in self._client.data:
+            return None
+
+        target_temperature = self._client.data["main_temperature_offset"]
+
+        if target_temperature is None:
+            return None
+
+        return target_temperature / 10
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        if "rh_sensor" not in self._client.data:
+            return None
+
+        return self._client.data["rh_sensor"]
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        if "main_airflow" not in self._client.data:
+            return None
+
+        if self._client.data["main_airflow"] is None:
+            return None
+
+        fan_mode = int(self._client.data["main_airflow"])
+        ha_fan_mode = SAVECAIR_FAN_MODES[fan_mode]
+
+        return ha_fan_mode
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        mode = self._client.get_current_operation()
+        if mode is None:
+            return None
+
+        return SAVECAIR_OPERATION_MODES[mode]
+
+    @property
+    def is_on(self):
+        """Return true if the device is on."""
+        if "main_airflow" not in self._client.data:
+            return True
+
+        if self._client.data["main_airflow"] is None:
+            return False
+        airflow = self._client.data["main_airflow"]
+
+        if airflow is not None:
+            return int(self._client.data["main_airflow"]) - 1 > 0
+
+        return False
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return OPERATION_LIST
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        if self.current_operation != STATE_MANUAL:
+            return []
```

Comment:
Why return an empty list?

Reply:
I received feedback that when fan_list returned None, HA spammed the log with "fan_list does not exists". They utilized an input_select, and read the fan_list. I figured one solution would be to return a empty list to remedy this. Thoughts?

Reply:
I can off course revert this, if it does not comply.

Reply:
If the entity doesn't support `fan_list` it should return `None` for that property (and not support that feature, but features can't change during the entity lifetime either).

Templates always need to handle non existing attributes. So I don't think we should change the property for that reason.